src/zenml/integrations/azure/orchestrators/azureml_orchestrator.py

```diff
@@ -0,0 +1,455 @@
+                if schedule.start_time is not None:
+                    start_time = schedule.start_time.isoformat()
+
+                end_time = None
+                if schedule.end_time is not None:
+                    end_time = schedule.end_time.isoformat()
+
+                if schedule.cron_expression:
+                    # If we are working with a cron expression
+                    schedule_trigger = CronTrigger(
+                        expression=schedule.cron_expression,
+                        start_time=start_time,
+                        end_time=end_time,
+                        time_zone=TimeZone.UTC,
+                    )
+
+                elif schedule.interval_second:
+                    # If we are working with intervals
+                    interval = schedule.interval_second.total_seconds()
+
+                    if interval < 3600:
+                        frequency = "minute"
+                        interval = int(interval // 60)
+                    elif interval < 86400:
+                        frequency = "hour"
+                        interval = int(interval // 3600)
+                    else:
+                        frequency = "day"
+                        interval = int(interval // 86400)
+
+                    recurrence_pattern = None
+                    if frequency == "day":
+                        if schedule.start_time is not None:
+                            recurrence_pattern = RecurrencePattern(
+                                hours=schedule.start_time.hour,
+                                minutes=schedule.start_time.minute,
+                            )
+                        else:
+                            recurrence_pattern = RecurrencePattern(
+                                hours=datetime.datetime.now().hour,
+                                minutes=datetime.datetime.now().minute,
+                            )
+
+                    schedule_trigger = RecurrenceTrigger(
+                        frequency=frequency,
+                        interval=interval,
+                        schedule=recurrence_pattern,
+                        start_time=start_time,
+                        end_time=end_time,
+                        time_zone=TimeZone.UTC,
+                    )
+
+                if schedule_trigger:
+                    # Create and execute the job schedule
+                    job_schedule = JobSchedule(
+                        name=run_name,
+                        trigger=schedule_trigger,
+                        create_job=pipeline_job,
+                    )
+                    ml_client.schedules.begin_create_or_update(
+                        job_schedule
+                    ).result()
+                    logger.info(
+                        f"Scheduled pipeline '{run_name}' with recurrence "
+                        "or cron expression."
+                    )
+                else:
+                    logger.warning(
```

Comment:
Both this warning and the error log below should be exceptions. Some of our downstream code relies on this call raising an exception if it fails, to for example delete some placeholder objects that we create.

Reply:
You are right. I have changed them into `RuntimeError`s now. Do we need a specific error here or are `RuntimeError`s just fine?

Reply:
RuntimeError is good